traitlets/traitlets.py

```diff
@@ -408,11 +410,21 @@ def _dynamic_default_callable(self, obj):
         If neither exist, it returns None
         """
         # Traitlets without a name are not on the instance, e.g. in List or Union
+
         if self.name:
             mro = type(obj).mro()
-            meth_name = '_%s_default' % self.name
+            for cls in mro[:mro.index(self.this_class)+1]:
+                for key in dir(cls):
+                    value = getattr(cls, key, None)
+                    if value and isinstance(value, DefaultHandler):
+                        if self.name in value.names:
+                            return getattr(obj, event_handler.name)
```

Comment:
@SylvainCorlay removed the need for storage of `EventHandlers` on `TraitType` instances and allowed multiple trait names in the default decorator as in observe and validate.